### PR TITLE
Detect Alpine package name with alias

### PR DESCRIPTION
**Pull request by at-wat**
Address https://github.com/ros/rosdistro/pull/31565#issuecomment-1000664152

On the recent version of Alpine including `edge`, [package `boost` does not exist](https://pkgs.alpinelinux.org/packages?name=boost&branch=v3.15&arch=x86_64) but [`boost1.77` package provides boost with alias name of `boost`](https://git.alpinelinux.org/aports/tree/main/boost1.77/APKBUILD#n71).
This PR fixes package detection when the package is installed by alias name.

Local test using docker:
```shell
$ docker run -it --rm -v $(pwd):/work/rosdep alpine:3.15
$ apk add py3-pip
$ python3 -m pip install /work/rosdep
$ rosdep init
$ sed -i -e 's|ros/rosdistro/master|alpine-ros/rosdistro/alpine-custom-apk|' /etc/ros/rosdep/sources.list.d/20-default.list
$ rosdep update
$ mkdir -p /work/tmp
$ cat << EOF > /work/tmp/package.xml
<package format="2">
  <name>dummy</name>
  <version>0.0.0</version>
  <description>foo</description>
  <license>dummy</license>
  <maintainer email="dummy@dummy.dummy">dummy</maintainer>
  <depend>libboost-atomic</depend>
</package>
EOF
$ rosdep install --from-path /work/tmp -y
WARNING: ROS_PYTHON_VERSION is unset. Defaulting to 3
executing command [apk add boost]
(1/34) Installing boost1.77-atomic (1.77.0-r1)
...
(34/34) Installing boost1.77 (1.77.0-r1)
Executing busybox-1.34.1-r3.trigger
OK: 85 MiB in 87 packages
#All required rosdeps installed successfully
```

### CODE REVIEW

**Review by ivanpauno — APPROVED** [review]
LGTM!

This needs a review from someone more familiar with the code though.

File: src/rosdep2/platforms/alpine.py [hunk 1]
```diff
@@ -61,11 +61,35 @@ def apk_detect(pkgs, exec_fn=read_stdout):
...
+    #    $
+    replaced_packages = []
+    reading = False
+    for line in exec_fn(cmd_replaces).splitlines():
+        if line.endswith(' replaces:'):
+            reading = True
+        elif line == '':
+            reading = False
+        elif reading:
+            replaced_packages.append(line)
```
[the remark this answers is not in the document]
Comment (at-wat) [reply src/rosdep2/platforms/alpine.py:90]: `apk` doesn't add whitespaces at now but adding strip shouldn't have negative effect.

**Comment by hidmic** [comment]
@cottsay friendly ping

**Review by cottsay — APPROVED** [review]
Thanks for submitting this patch. I did some testing, and this seems to work as expected.

I'll note that using the `... replaces:` doesn't perfectly align with the 'alias' concept we're looking for, for example, `apk info --replaces mesa` lists `mesa-dricore`, but `apk add mesa-dricore` doesn't work.

As far as I can tell, it shouldn't be a problem if non-aliases are listed by `apk_detect`. Just something to note.

**Pull request merged**